Automattic/jetpack
# Packages: Finish the constants package

This PR is an attempt to make the constants package a properly namespaced package with independent tests, and to have the rest of Jetpack use that package the new way, without breaking Jetpack or the existing tests.

#### Changes proposed in this Pull Request:
* Add a namespace to the constants package.
* Update the constants package to use PSR-4.
* Fix all the phpcs errors in the constants package and add some inline docs.
* Add a constants package README with examples.
* Move the existing constant tests to the package.
* Add the infrastructure to automatically test the constants package.
* Update the rest of Jetpack to use the new constants package.
* Update composer.lock with the new changes.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Part of the Jetpack DNA project - p1HpG7-70O-p2

#### Testing instructions:
* Checkout this branch.
* Run `rm -rf ./vendor` inside Jetpack.
* Run `composer install` inside Jetpack.
* Navigate to `packages/constants`
* Run `composer phpunit`.
* Verify that runs `composer install` internally, and then runs the package tests.
* Verify the package tests pass.
* Run `composer phpunit` again.
* Verify the second time is fast because the dependencies are already installed.
* Verify CI passes on this PR and 100% is green.
* Smoke test Jetpack to verify it still works well in both the admin and the frontend.

#### Proposed changelog entry for your changes:
* Finish the constants package and make its tests independent.
